### PR TITLE
Blog/js tracker 2.13.0

### PR DESCRIPTION
**Pull request by paulboocock**

### CODE REVIEW

File: _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+5. [Removing user fingerprinting](#user-fingerprint)
+6. [Updates and bug fixes](#updates)
+7. [Upgrading](#upgrade)
+8. [Documentation and help](#doc)
+
+<!--more-->
+
+<h2 id="edge-analytics">1. Edge Analytics with activity tracking callback</h2>
+
+[Activity Tracking][activity-tracking] is a popular feature of the Snowplow JavaScript tracker, however to achieve accurate activity timing it is common to configure the JavaScript tracker to send page ping events to the Snowplow collector quite frequently. Often the tracker is configured to send an event every 10 seconds, allowing acitivty timing to be accurate to 10 seconds. The side effect of increased accuracy, and activity tracking in general, is potentially a significant increase in events being sent to the Snowplow collector.
```
Comment (LyubaGolovina) [inline _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md]:
```suggestion
[Activity Tracking][activity-tracking] is a popular feature of the Snowplow JavaScript tracker, however to achieve accurate activity timing it is common to configure the JavaScript tracker to send page ping events to the Snowplow collector quite frequently. Often the tracker is configured to send an event every 10 seconds, allowing activity timing to be accurate to 10 seconds. The side effect of increased accuracy, and activity tracking in general, is potentially a significant increase in events being sent to the Snowplow collector.
```

File: _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+Version 2.13.0 of the JavaScript tracker introduces a new edge analytics feature that will allow the page ping events to be collected and aggregated in the browser before being sent to the collector in a single event. This requires some additional work on the part of the site maintainer as there are many ways you may choose to aggregate and send this information to the Snowplow collector.
+
+To achieve this, the JavaScript tracker now exposes a new function called `enableActivityTrackingCallback`. Using this callback means that each time a page ping event would normally be sent to the Snowplow collector instead your callback function will be executed with the page activity information.
+
+Below is a complete example of utilising this new functionality to aggregate page acitivty information and send the event as the page unloads:
```
Comment (LyubaGolovina) [inline _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md]:
```suggestion
Below is a complete example of utilising this new functionality to aggregate page activity information and send the event as the page unloads:
```

File: _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+With version 2.13.0, when a second (or third, etc) page view is tracked, the activity tracking timing will reset as though this is a brand new reload of the tracker. Using the same example above, this means if a second page view event is created when you are 5 seconds into the first page view, then the activity tracking will fire 10 seconds after the second page view; 15 seconds after the first page view.
+
+<h2 id="beacon-improvements">3. Improving Beacon API support</h2>
+
+The Beacon API functionality that was fully introduced in [version 2.10.2][2.10.0-blog] of the JavaScript tracker has proved a useful as a way of ensuring events are sent more reliably, as the sending of events is passed to the browser to handle asynchronously. This means events are sent even in the cases when the page is closed.
```
Comment (LyubaGolovina) [inline _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md]:
```suggestion
The Beacon API functionality that was fully introduced in [version 2.10.2][2.10.0-blog] of the JavaScript tracker has proved useful as a way of ensuring events are sent more reliably, as the sending of events is passed to the browser to handle asynchronously. This means events are sent even in the cases when the page is closed.
```

File: _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+Unfortauntely, this is quite a immature API and browser support has been hit and miss. We have received reports of the Beacon API not performing as expected on Chromium based browsers (Chrome, Opera, Edge) and Safari. With regards to Chromium browsers, this is a [known bug in Chromium][chrome-beacon-bug] that prevents Beacon from working however the JavaScript tracker successfully falls back to sending events as standard Post requests. This bug has very recently been fixed in Chromium and is now available in Chrome Canary, we expect to see this fix make it into Chrome Stable in a couple of months which will allow users to see the benefits of Beacon in Chrome too.
```
Comment (LyubaGolovina) [inline _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md]:
```suggestion
Unfortunately, this is quite an immature API and browser support has been hit and miss. We have received reports of the Beacon API not performing as expected on Chromium based browsers (Chrome, Opera, Edge) and Safari. With regards to Chromium browsers, this is a [known bug in Chromium][chrome-beacon-bug] that prevents Beacon from working however the JavaScript tracker successfully falls back to sending events as standard Post requests. This bug has very recently been fixed in Chromium and is now available in Chrome Canary, we expect to see this fix make it into Chrome Stable in a couple of months which will allow users to see the benefits of Beacon in Chrome too.
```

File: _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+3. The presence of fingerprinting technology may lead to trackers being blocked, disadvantaging companies who are using the Snowplow technology who are being responsible and transparent with visitors to the website about what data they are collecting and how they are using it.
+
+With this release all fingerprinting capabilities have been removed from the JavaScript tracker. The API methods still exist to ensure errors do not occur if upgrading but you will see warnings in the Developer Console of the browser and the User Fingerprint field of an event will no longer be populated. The fingerprinting API methods will be entirely removed in version 3.0.0.
+
+There are a couple of core concerns that have led us to the decision to remove fingerprinting:
+
+As part of this effort we have  removed the Augur Identity automatic context which utilised finger printing to identify users. If you were using this context and wish to continue, we suggest you manually attach the Augur Identity context using the trackers global contexts feature to maintain existing behaviour.
```
Comment (LyubaGolovina) [inline _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md]:
```suggestion
As part of this effort we have removed the Augur Identity automatic context which utilised fingerprinting to identify users. If you were using this context and wish to continue, we suggest you manually attach the Augur Identity context using the trackers global contexts feature to maintain existing behaviour.
```

File: _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+- Fix al.optimizely.get is not a function error ([#619][619])
+- Further harden the Optimizely integrations ([#654][654])
+- Use local sp.js for example pages ([#790][790])
+- Change deprecation strings to constants and reuse ([#791][791])
+
+<h2 id="upgrade">7. Upgrading</h2>
+
+The tracker is available as a published asset in the [2.13.0 Github release][2.13.0-tag]:
+
+To upgrade, host the 2.13.0 version of `sp.js` asset on a CDN, and load the tracker from there.
```
Comment (LyubaGolovina) [inline _posts/2020-01-29-snowplow-javascript-tracker-2.13.0-released/index.md]:
```suggestion
To upgrade, Snowplow Insights and Open Source users should host the 2.13.0 version of `sp.js` asset on a CDN, and load the tracker from there.
```

**Comment by paulboocock** [comment]
Ready to go once 2.13.0 of the JS Tracker is released.

**Pull request merged**